scipy/cluster/_hierarchy.pyx

```diff
@@ -1029,7 +1029,7 @@ def slink(double[:] dists, int n):
                 M[Pi[j]] = min(M[Pi[j]], M[j])
 
         for j in range(i):
-            if Lambda[j] >= Lambda[Pi[j]]:
+            if Lambda[j] > Lambda[Pi[j]]:
```

Comment:
Looks like the same change should be made for the similar inequality check above (by the symmetry/equivalence argument). It certainly doesn't break anything, but still I haven't figured out how it works exactly.